src/Glpi/Altcha/AltchaManager.php

```diff
@@ -126,21 +126,66 @@ private function __construct()
         $this->initSessionStorage();
     }
 
+    public function isEnabled(): bool
+    {
+        if (defined('GLPI_ALTCHA_ENABLED') && is_bool(GLPI_ALTCHA_ENABLED)) {
+            return GLPI_ALTCHA_ENABLED;
+        }
+
+        return true;
+    }
+
+    public function shouldStartWidgetOnLoad(): bool
+    {
+        if (
+            defined('GLPI_ALTCHA_WIDGET_AUTOSTART')
+            && is_bool(GLPI_ALTCHA_WIDGET_AUTOSTART)
+        ) {
+            return GLPI_ALTCHA_WIDGET_AUTOSTART;
+        }
+
+        return true;
+    }
+
+    public function shouldHideWidget(): bool
+    {
+        if (
+            defined('GLPI_ALTCHA_WIDGET_HIDE')
+            && is_bool(GLPI_ALTCHA_WIDGET_HIDE)
+        ) {
+            return GLPI_ALTCHA_WIDGET_HIDE;
+        }
+
+        return true;
+    }
```

Comment:
What is the functional impact? The methods and constants names should reflect the behaviour on user side.
IMHO, we should not be too permissive on the proposed behaviours:
 - can be enabled or not;
 - can be silent or not.

Reply:
He will need to manually click the checkbox

Reply:
OK so it is not compatible with the usage of `GLPI_ALTCHA_WIDGET_HIDE`. Maybe we should have only one constant for the behaviour configuration.
`GLPI_CAPTCHA_MODE`:
 - `disabled`: no CAPTCHA,
 - `hidden`: no user interaction required and hidden,
 - `auto`: no user interaction required and visible,
 - `interactive`: user interaction required (and therefore visible).

`ALTCHA` -> `CAPTCHA` as we may change the technical component we use in the future.

Reply:
Note that we will have to document this, so it have to be easy to document.